<launch>

	<node pkg="laserscan_to_pointcloud" type="laserscan_to_pointcloud_assembler" name="laserscan_to_pointcloud_assembler_octomap" output="screen">
		<param name="laser_scan_topic" type="str" value="tilt_scan" />
		<param name="pointcloud_publish_topic" type="str" value="pointcloud_laser_frame" />
		<param name="number_of_scans_to_assemble_per_cloud" type="int" value="1" />
		<param name="timeout_for_cloud_assembly" type="double" value="1.0" />
		<param name="target_frame" type="str" value="hokuyo_tilt_laser_link" />
		<param name="min_range_cutoff_percentage_offset" type="double" value="5.00" />
		<param name="max_range_cutoff_percentage_offset" type="double" value="0.95" />
		<param name="include_laser_intensity" type="bool" value="false" />
		<param name="interpolate_scans" type="bool" value="false" />
	</node>

	<node pkg="octomap_server" type="octomap_server_node" name="octomap_server">
		<param name="frame_id" type="string" value="map" />
		<param name="resolution" value="0.01" />
		<param name="base_frame_id" type="string" value="base_footprint" />
		<param name="height_map" type="bool" value="true" />
		<param name="sensor_model/max_range" type="double" value="10.0" />
		<param name="sensor_model/hit" type="double" value="0.7" />
		<param name="sensor_model/miss" type="double" value="0.3" />
		<param name="sensor_model/min" type="double" value="0.12" />
		<param name="sensor_model/max" type="double" value="0.97" />
		<param name="latch" type="bool" value="false" />
		<param name="filter_ground" type="bool" value="false" />
		<param name="ground_filter/distance" type="double" value="0.01" />
		<param name="ground_filter/angle" type="double" value="0.15" />
		<param name="ground_filter/plane_distance" type="double" value="0.05" />
		<param name="pointcloud_min_z" type="double" value="0.0" />
		<param name="pointcloud_max_z" type="double" value="3.0" />
		<param name="occupancy_min_z" type="double" value="0.0" />
		<param name="occupancy_max_z" type="double" value="3.0" />
		<remap from="cloud_in" to="pointcloud_laser_frame" />
	</node>
</launch>
